### PR TITLE
Expose ScriptCreateDialog to EditorPlugin

### PR DESCRIPTION
**Pull request by willnationsdev**
Solves #19458.

expose_script_create_dialog_2
expose_script_create_dialog_1

Other issues...

1. The XML documentation doesn't want to seem to update the names/descriptions of the parameters in the config function. That is, if I generate the file and then modify it so it's filled, those values and others revert back to how they were beforehand when I run the doctool. Not sure if that's all a bug or if I'm just putting information in wrong. I didn't bother filling in the documentation until this problem is solved.

2. ~I created an `/editor/icons/icon_script_create_dialog.svg` file. It is a copy of the `icon_script.svg` file, but with the added InkScape filter to Colorize it so that it matches the color of a Control node. However, even after recompiling, the Classes API dialog that searches through types shows the node in its original color. Any idea as to how to fix it?~ Edit: I solved this problem by directly editing the fill color of the objects in the original, unoptimized svg files from the godot-design repo.

expose_script_create_dialog_3

### CODE REVIEW

**Comment by willnationsdev** [comment]
cc @mhilbrunner regarding doc issues (problem 1).

**Comment by mhilbrunner** [comment]
Thats indeed weird, thanks for the CC. Maybe @akien-mga has an idea why the docs tool does that?
I will also have a look later, maybe I can find out. :)

**Comment by toger5** [comment]
> but with the added InkScape filter to Colorize it

I don't entirely understand what you mean by that?
But to make icons coloured you need to assign the exact rub value so it that the color get replaced for dark and light icon themes. So a filter (don't know what that is?) would not help since only the color of shapes get replaced.
check 'editor_themes.cpp'
(did you try to replace the icon with just a square? to make sure that the new one is loaded?)

**Comment by willnationsdev** [comment]
@toger5 I had previously created the green coloration using the Inkscape's Filter -> Color -> Colorize effect. I have now recreated the image by directly editing the Fill colors of the objects in the unoptimized svg file, and that seems to have fixed it. I also ran the optimization script in the godot-design repo and have copied the optimized version to this PR. Will be submitting a PR to the godot-design repo as with the update as the instructions dictate.

File: editor/editor_plugin.h [hunk 1]
```diff
@@ -31,6 +31,7 @@
 #ifndef EDITOR_PLUGIN_H
 #define EDITOR_PLUGIN_H
 
+#include "../editor/script_create_dialog.h"
```
Comment (akien-mga) [inline editor/editor_plugin.h]: No need for a relative path.

File: editor/script_create_dialog.cpp [hunk 2]
```diff
@@ -582,6 +582,9 @@ void ScriptCreateDialog::_bind_methods() {
 	ClassDB::bind_method("_path_changed", &ScriptCreateDialog::_path_changed);
 	ClassDB::bind_method("_path_entered", &ScriptCreateDialog::_path_entered);
 	ClassDB::bind_method("_template_changed", &ScriptCreateDialog::_template_changed);
+
+	ClassDB::bind_method("config", &ScriptCreateDialog::config);
```
Comment (akien-mga) [inline editor/script_create_dialog.cpp]: If you want the arguments named in the doc, you need to name them here.

**Comment by akien-mga** [comment]
I don't think we have any editor-specific node exposed as a builtin node type to instantiate yet... It seems quite weird to have this exposed in the Add Node dialog.

**Comment by willnationsdev** [comment]
Thanks for the feedback. I'll do that stuff as soon as I can.

The reason the EditirFileDialog was never at risk of showing up in the CreateDialog's list of nodes is because the CreateDialog specifically filters out any nodes with a class starting with "Editor".

    if (base_type == "Node" && type.begins_with("Editor"))
    
    continue; // do not show editor nodes

Because the ScriptCreateDialog is not the EditorScriptCreateDialog, you have to manually filter it out.

**Comment by akien-mga** [comment]
> The reason the EditirFileDialog was never at risk of showing up in the CreateDialog's list of nodes is because the CreateDialog specifically filters out any nodes with a class starting with "Editor".

Isn't this what your `type_blacklist` aims to do?

**Comment by willnationsdev** [comment]
@akien-mga Yes

File: editor/script_create_dialog.cpp (hunk 2 above)
Comment (willnationsdev) [reply editor/script_create_dialog.cpp]: Ahhh, thank you. Didn't catch this.

**Comment by willnationsdev** [comment]
@akien-mga This PR should be good to go now.

**Pull request merged**